File: javascript/gemini/p56.js
app.post('/change_email', (req, res) => {
    if (!req.session.user) return res.status(403).send("Login required");

    const { oldEmail, newEmail, password } = req.body;
    
    // Verify password & old email logic here
    if (newEmail === oldEmail) {
        return res.send("New email must be different");
    }

    db.updateEmail(req.session.user.id, newEmail, password, (err) => {
        if (err) res.send("Error or Wrong Password");
        else res.send("Success");
    });
});

File: python/gemini/p56.py
def change_email(user, old_email, new_email, password):
    if not user.is_authenticated:
        return "Not logged in"

    if not user.verify_password(password):
        return "Incorrect password"

    if user.email != old_email:
        return "Old email mismatch"

    if old_email == new_email:
        return "New email must be different"

    user.email = new_email
    user.save()
    return "Success"
